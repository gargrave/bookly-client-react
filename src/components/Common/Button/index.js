import React from 'react'
import { func, string } from 'prop-types'

const classname = (type) => {
  if (type === 'success') {
    return 'buttons-success'
  }
}

const Button = ({
  onClick,
  text,
  type,
}) => {
  return (
    <button
      className={classname(type)}
      style={{ marginLeft: '10px' }}
      onClick={onClick}>
      { text }
    </button>
  )
}

Button.propTypes = {
  onClick: func.isRequired,
  text: string.isRequired,
  type: string.isRequired,
}

export default Button
