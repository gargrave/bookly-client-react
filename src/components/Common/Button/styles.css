.bookly-button {
  margin-left: 10px;
}
