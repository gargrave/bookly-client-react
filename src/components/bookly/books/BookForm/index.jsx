import React from 'react'
import { array, func, object, shape, string } from 'prop-types'

import AuthorSelect from '@/components/bookly/authors/AuthorSelect'

const BookForm = ({ authors, book, handleAuthorChange, handleCancel, handleInputChange, handleSubmit }) => {
  return (
    <form
      className="author-form"
      style={{ maxWidth: '500px', margin: 'auto', marginTop: '20px', textAlign: 'left' }}
      onSubmit={handleSubmit}
      noValidate
    >
      <div className="input-single">
        <label htmlFor="title">Title:</label>
        <input type="text" name="title" id="title" value={book.title} onChange={handleInputChange} />
      </div>

      <AuthorSelect author={book.author} authors={authors} handleChange={handleAuthorChange} />

      <div className="input-single">
        <button type="submit" className="submit-button" onClick={handleSubmit}>
          Submit
        </button>
        <button type="button" className="button-info float-right cancel-button" onClick={handleCancel}>
          Cancel
        </button>
      </div>
    </form>
  )
}

BookForm.propTypes = {
  book: shape({
    title: string.isRequired,
    author: object.isRequired,
  }).isRequired,
  authors: array.isRequired,
  handleAuthorChange: func.isRequired,
  handleInputChange: func.isRequired,
  handleSubmit: func.isRequired,
  handleCancel: func.isRequired,
}

export default BookForm
