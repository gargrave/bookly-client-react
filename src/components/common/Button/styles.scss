.bookly-button {
  &:not(&--left) {
    margin-left: 10px;
  }
}
