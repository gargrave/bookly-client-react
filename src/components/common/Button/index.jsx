// @flow
import React from 'react';
import { bool, func, oneOf, string } from 'prop-types';

import { buildClasses } from '../../../utils/cssHelpers';

import './styles.css';

type Props = {
  canSubmit?: boolean,
  classes?: string,
  onClick: Function,
  position?: string,
  text: string,
  type: string,
};

const acceptableTypes = [
  'success',
  'secondary',
  'info',
  'warning',
  'danger',
  'light',
  'dark',
];

const acceptablePositions = [
  'left',
];

function buttonClass(type: string) {
  if (acceptableTypes.includes(type)) {
    return `button-${type}`;
  }
  return '';
}

function positionClass(position: string) {
  if (acceptablePositions.includes(position)) {
    return `button-${position}`;
  }
  return '';
}

function buildClassList(
  type: string,
  position: string,
  classes: string,
) {
  return buildClasses(
    ['button', positionClass(position)],
    [buttonClass(type), ...classes.split(' ')]
  );
}

function Button({
  canSubmit,
  classes = '',
  onClick,
  position = '',
  text,
  type,
}: Props) {
  return (
    <button
      type={canSubmit ? 'submit' : 'button'}
      className={buildClassList(type, position, classes)}
      onClick={onClick}>
      { text }
    </button>
  );
}

Button.propTypes = {
  canSubmit: bool,
  classes: string,
  onClick: func.isRequired,
  position: oneOf(acceptablePositions),
  text: string.isRequired,
  type: oneOf(acceptableTypes).isRequired,
};

export default Button;
