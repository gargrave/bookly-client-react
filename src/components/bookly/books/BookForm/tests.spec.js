import React from 'react'
import { shallow } from 'enzyme'

import Button from '@/components/common/Button'

import BookForm from './'

describe('BookForm', () => {
  let props
  let component

  describe('with "book" populated', () => {
    beforeEach(() => {
      props = {
        book: { title: '', author: {} },
        authors: [],
        onAuthorChange: jest.fn(),
        onInputChange: jest.fn(),
        onSubmit: jest.fn(),
        onCancel: jest.fn(),
      }

      component = shallow(<BookForm {...props} />)
    })

    test('renders correctly', () => {
      expect(component).toMatchSnapshot()
      expect(component.find('form.bookly-form').length).toEqual(1)
      expect(component.find('form.bookly-book-form').length).toEqual(1)
      expect(component.find('input[type="text"]').length).toEqual(1)
      expect(component.find(Button).length).toEqual(2)
    })
  })
})
