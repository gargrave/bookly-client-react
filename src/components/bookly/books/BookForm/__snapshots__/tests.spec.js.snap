// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookForm with "book" populated renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="bookly-form bookly-book-form"
    noValidate={true}
    onSubmit={[Function]}
>
    <div
        className="input-field"
    >
        <label
            htmlFor="title"
        >
            Title:
        </label>
        <input
            id="title"
            name="title"
            onChange={[Function]}
            type="text"
            value=""
        />
    </div>
    <AuthorSelect
        author={Object {}}
        authors={Array []}
        onChange={[Function]}
    />
    <div
        className="input-field"
    >
        <Button
            canSubmit={true}
            onClick={[Function]}
            position="left"
            text="Submit"
            type="success"
        />
        <Button
            extraClasses="float-right"
            onClick={[Function]}
            text="Cancel"
            type="light"
        />
    </div>
</form>,
  "nodes": Array [
    <form
      className="bookly-form bookly-book-form"
      noValidate={true}
      onSubmit={[Function]}
>
      <div
            className="input-field"
      >
            <label
                  htmlFor="title"
            >
                  Title:
            </label>
            <input
                  id="title"
                  name="title"
                  onChange={[Function]}
                  type="text"
                  value=""
            />
      </div>
      <AuthorSelect
            author={Object {}}
            authors={Array []}
            onChange={[Function]}
      />
      <div
            className="input-field"
      >
            <Button
                  canSubmit={true}
                  onClick={[Function]}
                  position="left"
                  text="Submit"
                  type="success"
            />
            <Button
                  extraClasses="float-right"
                  onClick={[Function]}
                  text="Cancel"
                  type="light"
            />
      </div>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BookForm
        authors={Array []}
        book={
                Object {
                        "author": Object {},
                        "title": "",
                      }
        }
        onAuthorChange={[Function]}
        onCancel={[Function]}
        onInputChange={[Function]}
        onSubmit={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "authors": Array [],
          "book": Object {
            "author": Object {},
            "title": "",
          },
          "onAuthorChange": [Function],
          "onCancel": [Function],
          "onInputChange": [Function],
          "onSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="bookly-form bookly-book-form"
          noValidate={true}
          onSubmit={[Function]}
>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="title"
                    >
                              Title:
                    </label>
                    <input
                              id="title"
                              name="title"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </div>
          <AuthorSelect
                    author={Object {}}
                    authors={Array []}
                    onChange={[Function]}
          />
          <div
                    className="input-field"
          >
                    <Button
                              canSubmit={true}
                              onClick={[Function]}
                              position="left"
                              text="Submit"
                              type="success"
                    />
                    <Button
                              extraClasses="float-right"
                              onClick={[Function]}
                              text="Cancel"
                              type="light"
                    />
          </div>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="bookly-form bookly-book-form"
          noValidate={true}
          onSubmit={[Function]}
>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="title"
                    >
                              Title:
                    </label>
                    <input
                              id="title"
                              name="title"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </div>
          <AuthorSelect
                    author={Object {}}
                    authors={Array []}
                    onChange={[Function]}
          />
          <div
                    className="input-field"
          >
                    <Button
                              canSubmit={true}
                              onClick={[Function]}
                              position="left"
                              text="Submit"
                              type="success"
                    />
                    <Button
                              extraClasses="float-right"
                              onClick={[Function]}
                              text="Cancel"
                              type="light"
                    />
          </div>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookForm
    authors={Array []}
    book={
        Object {
            "author": Object {},
            "title": "",
          }
    }
    onAuthorChange={[Function]}
    onCancel={[Function]}
    onInputChange={[Function]}
    onSubmit={[Function]}
/>,
}
`;
